nilearn/decoding/decoder.py

```diff
@@ -710,17 +720,28 @@ def fit(self, X, y, groups=None):
         n_final_features = int(
             X.shape[1]
             * self.screening_percentile_
-            * self.clustering_percentile
+            * self._clustering_percentile
             / 10000
         )
         if n_final_features < 50:
+            screening_percentile_msg = ""
+            if self.screening_percentile_ < 100:
+                screening_percentile_msg = (
+                    "Consider raising 'screening_percentile'"
+                )
+            clustering_percentile_msg = ""
+            if (
+                hasattr(self, "clustering_percentile")
+                and self._clustering_percentile < 100
+            ):
+                clustering_percentile_msg = " and / or 'clustering_percentile'"
+            warning_msg = (
+                "The decoding model will be trained only "
+                f"on {n_final_features} features. "
+                f"{screening_percentile_msg}{clustering_percentile_msg}."
+            )
```

Comment:
this mostly tries to refine the message:
- no point in mentioning things to raise above a 100%
- only mention clustering percentile for decoders that make use of it